### PR TITLE
Add .taskcluster.yml

### PR DESCRIPTION
**Pull request by ricky26**

### CODE REVIEW

**Comment by robot-gunslinger** [comment]
:hammer: [cloudbuild-job-z9k48](https://internal.wpgapi.uk/foreman/namespaces/infra-ci/jobs/cloudbuild-job-z9k48/) started.

**Comment by robot-gunslinger** [comment]
:hammer: [cloudbuild-job-c6zdg](https://internal.wpgapi.uk/foreman/namespaces/infra-ci/jobs/cloudbuild-job-c6zdg/) started.

**Comment by robot-gunslinger** [comment]
:heavy_check_mark: [cloudbuild-job-z9k48](https://internal.wpgapi.uk/foreman/namespaces/infra-ci/jobs/cloudbuild-job-z9k48/) succeeded. :tada:

| Metadata | Value |
| --- | --- |
| Cloud Build | [Download Logs](https://storage.googleapis.com/wpg-cloudbuild-tmp/log-a43f9615-b252-4790-b4b4-5cee26248e46.txt?Expires=1603288243&GoogleAccessId=cloudbuild%40wpg-internal.iam.gserviceaccount.com&Signature=0ZnZVC7QeaVPihNEscKDDoE7PofgRQB45gItGiwpoJw%2FV23SrioFS7JAlSNjGzhZWRJjsG0zZ%2FrTSnfhHS7cUYS%2BFZgmaMOatXU4kRJvJPTeFCI1e5eq1Grtx%2FhE%2FucVwJtbbUjxBlXTJf2%2BYSzzO6evxLDxsVYg1KUhrNGsT0WdAvEG8t1sy8EA2Qd6UJSLDI6UYvFE3U8NjrnnOvvmAml83Olghj0hNI5DUZw%2F3i%2FqsRnGYelOZCszrR7bRiX2kvv7s12pyDtnXegKQRNTEZvx0w4ZPKMWBCfDYlOHehcwZrUiyL6T6YJ%2FlNR5%2FGvgsBUpamXFzhrQ0eE5yjqYiQ%3D%3D) |
| Cloud Build Dashboard | [View Logs](https://console.cloud.google.com/cloud-build/builds/a43f9615-b252-4790-b4b4-5cee26248e46?project=293430010285) |

**Comment by robot-gunslinger** [comment]
:heavy_check_mark: [cloudbuild-job-c6zdg](https://internal.wpgapi.uk/foreman/namespaces/infra-ci/jobs/cloudbuild-job-c6zdg/) succeeded. :tada:

| Metadata | Value |
| --- | --- |
| Cloud Build | [Download Logs](https://storage.googleapis.com/wpg-cloudbuild-tmp/log-9d04c444-4bb0-4b9e-acc2-41572ec4caca.txt?Expires=1603288361&GoogleAccessId=cloudbuild%40wpg-internal.iam.gserviceaccount.com&Signature=aooFvOA8rpzeyr37%2FqSoja%2FBvnTxYlrU5n%2Bh6z0AD0s1ByDO5T1ykf%2BGab3ZF55quFte3Pm%2FBJZxflJPD46Ycd3m1PASgari7ZPfZgka%2FmXeIlnJzGcUKSVG2lFH5Cofyqctnr0NHaJkOSlya9fqqXU9SYZGo5BMy67STnxb%2F7REm3DL0CsOE%2BgcqojCPKwNqSTxnuhVUCObBwiXWkVuDpIdxBTSaLlKewjQc0YDfiz1wkrthhYU%2F8b4DooBVo3DqI0CTzlMHR5HJZwqcIziM4l6RUM9JcYyIDydVhSFIMCE%2FmRaEPtt0b0vCDYiQtNj8WQiEn1LC5KYOH0GjKUoRA%3D%3D) |
| Cloud Build Dashboard | [View Logs](https://console.cloud.google.com/cloud-build/builds/9d04c444-4bb0-4b9e-acc2-41572ec4caca?project=293430010285) |

**Comment by robot-gunslinger** [comment]
:hammer: [cloudbuild-job-4j676](https://internal.wpgapi.uk/foreman/namespaces/infra-ci/jobs/cloudbuild-job-4j676/) started.

**Comment by robot-gunslinger** [comment]
:heavy_check_mark: [cloudbuild-job-4j676](https://internal.wpgapi.uk/foreman/namespaces/infra-ci/jobs/cloudbuild-job-4j676/) succeeded. :tada:

| Metadata | Value |
| --- | --- |
| Cloud Build | [Download Logs](https://storage.googleapis.com/wpg-cloudbuild-tmp/log-9bf7cf4d-acea-48a8-9d70-3a2b6b3a3055.txt?Expires=1603290555&GoogleAccessId=cloudbuild%40wpg-internal.iam.gserviceaccount.com&Signature=vxUAdD9x55ZY7uvMF3iEpl4LRybZFgQWk1%2F%2BmmWz%2BqIDN3TLrSyMMFkpMoFMV5pkB%2FwU9aT9NHxZW8gLRd%2BTRm9tB2h68Ty8EBwtvcLhuLOcGc%2FZUQ2pe2snthRSaBUbI27Z3F2z2v6Hs9hAuPnyzATKPRsGbcQ3fC2qui1T6tVgqlYnnroOBx8tTRY66PCbbU%2B2lU%2FOz4hLsVgsIfZ7LTaHSFxYKSYCq9dwJP07C76QAd%2FGwV0MCCISn%2Bpl4fG7jUbD7zrpIGjFBCBfpWQq74ylHMlrDPZCDrYA9L8dO44IldfJs4GZD0Q93HFl38pIyqHnYXnO5pc5sKQBTeTnMw%3D%3D) |
| Cloud Build Dashboard | [View Logs](https://console.cloud.google.com/cloud-build/builds/9bf7cf4d-acea-48a8-9d70-3a2b6b3a3055?project=293430010285) |

File: .taskcluster.yml [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+                - image: bitnami/git
+                  command: ["/bin/bash", "-c"]
+                  args:
+                    - git clone ${event.repository.html_url} . && git checkout ${event.after}
+                - image: docker
+                  args: [ "build", ".", "-t", "${imageName}:${event.after}" ]
+                - image: docker
+                  args: [ "push", "${imageName}:${event.after}" ]
+                - image: gcr.io/wpg-internal/update-infra-image
+                  args: ["deployments/packages/_taskcluster-operator/package.json=config.imageTags.taskcluster-operator=${event.after}"]
```
Comment (omaskery) [inline .taskcluster.yml:37]: what the hecking is this argument syntax

File: .taskcluster.yml [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+              - image: gcr.io/wpg-internal/ci-bootstrap
+              - image: bitnami/git
+                command: ["/bin/bash", "-c"]
+                args:
+                  - |
+                    git clone ${event.repository.html_url}
+                    && git checkout ${event.pull_request.head.sha}
+                    && git rebase ${event.pull_request.base.sha}
+              - image: docker
+                args: [ "build", "." ]
```
Comment (omaskery) [inline .taskcluster.yml:65]: no push with the PR number, or is this something we're phasing out in preference of other future tooling?

**Review by omaskery — CHANGES_REQUESTED** [review]
discussions

File: .taskcluster.yml (hunk 1 above)
Comment (omaskery) [reply .taskcluster.yml:37]: let's see about noticing completed taskcluster tasks and having a service notice that and update infra for you by knowing the mapping from repos to images, and images to the packages that use them? :)

File: .taskcluster.yml (hunk 2 above)
Comment (ricky26) [reply .taskcluster.yml:65]: Nobody has ever used those images (unless you have)... I can't see much of an argument for keeping them...

Comment (omaskery) [reply .taskcluster.yml:65]: I technically have used them, but they immediately fall over on the fact infra/k8s can't detect changes in those images c cos they use the same image name. Technically it'd be handy for people who don't want to/cannot run my build&deploy script (or perform it manually like you do) to open a PR and use the tag to manually put in infra and do a scratch-2 deployment (for example). Not a hard requirement, just want people to have a nice workflow for this, ala forklift v2 or whatever, yadda yadda.

**Comment by robot-gunslinger** [comment]
:hammer: [cloudbuild-job-c5pnp](https://internal.wpgapi.uk/foreman/namespaces/infra-ci/jobs/cloudbuild-job-c5pnp/) started.

**Comment by robot-gunslinger** [comment]
:heavy_check_mark: [cloudbuild-job-c5pnp](https://internal.wpgapi.uk/foreman/namespaces/infra-ci/jobs/cloudbuild-job-c5pnp/) succeeded. :tada:

| Metadata | Value |
| --- | --- |
| Cloud Build | [Download Logs](https://storage.googleapis.com/wpg-cloudbuild-tmp/log-6383798b-93d6-46b1-8fa6-73ad55584ede.txt?Expires=1603907207&GoogleAccessId=cloudbuild%40wpg-internal.iam.gserviceaccount.com&Signature=gq2G7GOTKSKAvlzky7tqloW0a%2BHJPdi2KmmX465nkoCF1TUKZxIHvGw%2F3vA%2FBMjVlOlwFjHCMm6R6FV9ChrWe5AUSmIxxOITgbwHlenMkBLhpmFZwljaX%2BRbyXSTL5ZMJm6ycdDsHOBHANNSNX9rpmQBZhBkv7Gd%2FEOdMfOCxt3yU6CFM6BuvcHfWYkKlHmim%2FwKTjJzlBumtNY7UK3IgtsQ8GLreysJl2EWtQ9q3Ay4GcDstJdKLEWx8D83BsZuj%2BW7xa8O%2B1uY1rXrcsuEj4i3eVhcgGu2ROEcNs%2F1Tsx%2BwidS%2B%2FHCUEJzEv%2FNzb6hGwFOGIz%2B6cNt0pbbrSI%2Fkg%3D%3D) |
| Cloud Build Dashboard | [View Logs](https://console.cloud.google.com/cloud-build/builds/6383798b-93d6-46b1-8fa6-73ad55584ede?project=293430010285) |

**Review by omaskery — APPROVED** [review]

**Pull request merged**